webproject/src/main/java/com/google/sps/util/FirebaseUtil.java

```diff
@@ -0,0 +1,77 @@
+package com.google.sps.util;
+
+import java.io.FileInputStream;
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.ServletException;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.Query;
+import com.google.firebase.database.ValueEventListener;
+
+@Singleton
+public class FirebaseUtil {
+    @Inject
+    public FirebaseUtil() throws Exception {
+        // Fetch the service account key JSON file contents
+        FileInputStream serviceAccount = new FileInputStream("./key.json");
+
+        // Initialize the app with a service account, granting admin privileges
+        FirebaseOptions options = new FirebaseOptions.Builder()
+                .setCredentials(GoogleCredentials.fromStream(serviceAccount))
+                .setDatabaseUrl("https://summer20-sps-47.firebaseio.com")
+                .build();
+        
+        if (FirebaseApp.getApps().isEmpty()) {
+            FirebaseApp.initializeApp(options);
+        }
+    }
+
+    public DatabaseReference getUserRoomReference() {
+        return FirebaseDatabase.getInstance()
+            .getReference("UserRoom");
+    }
+
+    public DatabaseReference getRoomsReference() {
+        return FirebaseDatabase.getInstance()
+                .getReference("rooms");
+    }
+
+    public Optional<DataSnapshot> getQuerySnapshot(Query query, String input) throws ServletException {
+        final BlockingQueue<Optional<DataSnapshot>> queue = new LinkedBlockingDeque(1);
+        query.equalTo(input).addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                if (dataSnapshot.exists()) {
+                    queue.add(Optional.of(dataSnapshot));
+                } else {
+                    queue.add(Optional.empty());
+                }
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+                throw databaseError.toException();
+            }
+        });
+
+        try {
+            return queue.poll(30, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            throw new ServletException("The database did not return a response for the specified query");
+        }
+    }
+
```

Comment:
please remove extra newlines in all the files that are not necessary

Reply:
oops >.<